django/contrib/postgres/fields/array.py

```diff
@@ -88,6 +88,16 @@ def to_python(self, value):
             value = [self.base_field.to_python(val) for val in vals]
         return value
 
+    def from_db_value(self, value, expression, connection, context):
+        if value is None:
+            return value
+        return [
+            self.base_field.from_db_value(val, expression, connection, context)
+            if hasattr(self.base_field, 'from_db_value') else
+            self.base_field.to_python(val)
```

Comment:
According to Django docs:
> If present for the field subclass, from_db_value() will be called in all circumstances when the data is loaded from the database, including in aggregates and values() calls.
> to_python() is called by deserialization and during the clean() method used from forms.

https://docs.djangoproject.com/en/1.8/howto/custom-model-fields/#converting-values-to-python-objects